elytron/WFLY-11697_WS_REST_client_integration_with_Elytron.adoc

```diff
@@ -0,0 +1,60 @@
+= WFLY-11697 WS and Rest integration with WildFly Elytron - AuthenticationClient for Authentication / SSL
+:author:            Diana Vilkolakova
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          elytron, client, resteasy, webservices
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The wildfly-config.xml file is used on the client side of web apps and it contains the authentication configuration that should be used
+when attempting to connect to the server (username, password, ssl config, mechanism specification etc).
+It can either be included with the client application or specified using the -Dwildfly.config.url property when running the client.
+This requested feature is about making use of wildfly-config.xml file on the client side of WS / REST deployment.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11697[WFLY-11697]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1097[EAP7-1097]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolakova]
+
+=== QE Contacts
+
+* TBD
+
+=== Affected Projects or Components
+
+* REST
+* Security
+* Web Services
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+If wildfly-config.xml file is present when using RESTEasy or Web Services on the client side, this authentication configuration will be parsed and made available for this client.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Implementation Plan
+
+SPI will be exposed in RESTEasy and Web Services for client's authentication configuration. This SPI will be implemented in Elytron.
+
```

Comment:
Alternatively, this SPI will be in its own module that both RestEasy and WS depend on and Elytron will implement.

Reply:
@asoldano So RestEasy will have SPI exposed that Elytron client implements and the loading of service providers could happen during build of RestEasyClientBuilder. Does this sound reasonable?

Reply:
@Skyllarr I suggest dealing with JBossWS and RESTEasy independently. JBossWS has its own SPI component, jbossws-spi (https://github.com/jbossws/jbossws-spi), which really looks like where the additional spi should be defined.
RESTEasy doesn't have such clearly defined spi component, so I believe a new module will have to be added to the project.

Reply:
@asoldano Thank you for your reply! We will split this issue and will deal with JBossWS and RESTEasy separately.